src/components/views/beacon/RoomCallBanner.tsx

```diff
@@ -0,0 +1,134 @@
+import {
+    OwnBeaconStore,
+    OwnBeaconStoreEvent,
+} from "../../../stores/OwnBeaconStore";
+import { CallDurationFromEvent } from "../voip/CallDuration";
+
+interface RoomCallBannerProps {
+    roomId: Room["roomId"];
+    call: Call;
+}
+
+const RoomCallBannerInner: React.FC<RoomCallBannerProps> = ({
+    roomId,
+    call,
+}) => {
+    const callEvent: MatrixEvent | null = (call as ElementCall)?.groupCall;
+
+    const connect = useCallback(
+        (ev: ButtonEvent) => {
+            ev.preventDefault();
+            defaultDispatcher.dispatch<ViewRoomPayload>({
+                action: Action.ViewRoom,
+                room_id: roomId,
+                view_call: true,
+                metricsTrigger: undefined,
+            });
+        },
+        [roomId],
+    );
+
+    const onClick = useCallback(() => {
+        dispatcher.dispatch<ViewRoomPayload>({
+            action: Action.ViewRoom,
+            room_id: roomId,
+            metricsTrigger: undefined,
+            event_id: callEvent.getId(),
+            scroll_into_view: true,
+            highlighted: true,
+        });
+    }, [callEvent, roomId]);
+
+    if (!call) return <Fragment />;
+
+    return (
+        <div
+            className="mx_RoomCallBanner"
+            onClick={onClick}
+        >
+            <div className="mx_RoomCallBanner_text">
+                <span className="mx_RoomCallBanner_label">{ _t("Video call") }</span>
+                <CallDurationFromEvent mxEvent={callEvent} />
+            </div>
+
+            <AccessibleButton
+                onClick={connect}
+                kind="primary"
+                element="button"
+                disabled={false}
+            >
+                { _t("Join") }
+            </AccessibleButton>
+        </div>
+    );
+};
+
+interface Props {
+    roomId: Room["roomId"];
+}
+
+const RoomCallBanner: React.FC<Props> = ({ roomId }) => {
+    const call = useCall(roomId);
+
+    // this section is to check if we have a live location share. If so, we dont show the call banner
+    const isMonitoringLiveLocation = useEventEmitterState(
+        OwnBeaconStore.instance,
+        OwnBeaconStoreEvent.MonitoringLivePosition,
+        () => OwnBeaconStore.instance.isMonitoringLiveLocation,
+    );
+
+    const liveBeaconIds = useEventEmitterState(
+        OwnBeaconStore.instance,
+        OwnBeaconStoreEvent.LivenessChange,
+        () => OwnBeaconStore.instance.getLiveBeaconIds(roomId),
+    );
+
+    if (isMonitoringLiveLocation && liveBeaconIds.length) {
+        return null;
+    }
+
+    // Check if the call is already showing. No banner is needed in this case.
+    if (RoomViewStore.instance.isViewingCall()) {
+        return null;
+    }
+
+    // split into outer/inner to avoid watching various parts if there is no call
+    if (call) {
+        // No banner if the call is not connected
+        if (call.connectionState !== ConnectionState.Disconnected) {
```

Comment:
You should use `useConnectingState`. Also, I would personally prefer the if statement to be on one line

Reply:
I think useConnectionState does not work in this case since call might be null. (and I cannot setup the hook inside the conditional)